ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/resources/jts/orbspecific/XAResourceRecord.java

```diff
@@ -746,6 +761,9 @@ public void commit_one_phase() throws HeuristicHazard,
 	                    _theXAResource.commit(_tranID, true);
 	                else {
 	                    _theXAResource.rollback(_tranID);
+						// _rolledBack set to true even though
+						// it's not needed at this stage
+						this._rolledBack = true;
```

Comment:
Is it not set by 333/358? Is there another path?

Reply:
I'm not sure I understand this comment. Why would commit_one_phase depend on rollback?

Reply:
Sorry - I misunderstood. I will resolve this!